cmd/util/cmd/common/flow_client.go

```diff
@@ -139,3 +128,29 @@ func convertAccessAddrFromState(address string, insecureAccessAPI bool) string {
 
 	return accessAddress.String()
 }
+
+// DefaultAccessNodeIDS will return all the access node IDS in the protocol state for staked access nodes
+func DefaultAccessNodeIDS(snapshot protocol.Snapshot) ([]flow.Identifier, error) {
```

Comment:
1. This should probably never be allowed to return 0 IDs.
2. This can return < 2 IDs. Depending on whether we're in a test situation, and. whether the test is specifically exercising AN fallback mechanisms, this may or may not be a bad thing. But I would at the very least log a warning there.

Reply:
Add sanity check for 0 identities found https://github.com/onflow/flow-go/pull/1426/commits/9abd914a2ab87872409d7993616f732b480b1a8c 

As for 2. all tests explicitly enforce the access node ID minimum of 2 it's a requirement see [here](https://github.com/onflow/flow-go/blob/master/integration/testnet/network.go#L483)